client/mysqlbinlog.cc

```diff
@@ -143,6 +143,7 @@ static uint verbose= 0;
 
 static ulonglong start_position, stop_position;
 static const longlong stop_position_default= (longlong)(~(my_off_t)0);
+static ulonglong last_processed_position= 0;
```

Comment:
The comparison down there looks so wrong having to cast (C-style or `static_const`).
Will an unsigned do? Or even a this-file-only macro-constant?
```suggestion
static ulonglong start_position, stop_position;
#define stop_position_default ~0ULL
static ulonglong last_processed_position= 0;
```
or `ULLONG_MAX` or `std::numeric_limits<unsigned long long>::max()`…

---

Rather, specializing this value would put ~0 as an input out of support, though I don’t think it has a practical difference from “your file is bigger than the legal limit”.